### DISCUSSION TITLE
Problem with sorting products in a category by product placement in the category

### DISCUSSION THREAD

**Discussion by Kos-tash** [PrestaShop/PrestaShop#37728] [Q&A] [answered]
Hello everyone. I have a problem with nleft and nright. I'm looking for help from the community because I can't solve it myself.

I have a store with a certain number of categories and everything seems to be working fine. However, when I add or move a category, the sorting of products by category order stops working correctly in some categories.

I can't understand why this is happening and how to demonstrate this problem to me. But I will try to show a table from the database.

This video shows that everything is working correctly and normally. - [https://drive.google.com/file/d/1CAVpQC7x7IS5FxubEGPQIKMuR86aufQH/view?usp=sharing](https://drive.google.com/file/d/1CAVpQC7x7IS5FxubEGPQIKMuR86aufQH/view?usp=sharing)

and this is a table from the database - https://drive.google.com/file/d/1weE83FF1uEiNt2wEPmKFnqv6IJOO55uP/view?usp=sharing

Now let's try moving one category and look at the result.

One category started displaying products by ID from smallest to largest - https://drive.google.com/file/d/14CWqqpVlZ7H-dnxqFTVlmXZgwd8KAE6U/view?usp=sharing

And here is the table from the database. - https://drive.google.com/file/d/1SztV9C84SieDWMKPgu8Oh1-9bnaZnHlv/view?usp=sharing

Maybe someone can tell me what the problem is and how I can solve it.
I currently have Prestashop 8.2.0, but this problem has been going on since version 1.6.1.24.

**Comment by Kos-tash** [comment]
I noticed that even in the first version there are other categories with problems in sorting.
Does anyone else have such problems or do you not pay attention to sorting products in categories?

**Comment by Kos-tash** [comment]
Maybe there are some requirements for the category structure? Here is part of my category structure.

![image](https://github.com/user-attachments/assets/cfb2fb0a-84ce-4999-a60e-38b04033354c)

**Comment by musicpanda** [comment]
Your story sounds a bit confusing. The building of the nleft-nright tree uses only the position, depth and id_parent fields. If your category displays according to id that may be your setting for the category and unrelated to the position field.

**Comment by Kos-tash** [comment]
I absolutely agree with you. I can't solve this puzzle.
So far I've only found related changes to the nleft and nright fields with sorting in categories. 
`and this is a table from the database - https://drive.google.com/file/d/1weE83FF1uEiNt2wEPmKFnqv6IJOO55uP/view?usp=sharing`
In this variant, sorting by product location in the category works fine in category ID 4, but does not work in category ID 42.
`And here is the table from the database. - https://drive.google.com/file/d/1SztV9C84SieDWMKPgu8Oh1-9bnaZnHlv/view?usp=sharing`
And here sorting by product location in the category does not work in category ID 4, but works perfectly in category ID 42.
I can't understand how adding, moving, or editing any category affects the sorting of products in other categories.

I migrated my store from Prestashop 1.6.1.24 to 8.0.1. I have now upgraded to version 8.2, but the problem still persists.
I have a test store where there are all categories and a hundred products. And it has the same problems as in the main store. I thought that maybe there is a problem with the module, but when switching from 1.6 to 8.0 I did not use some of the modules, and some I took new versions that are not even compatible with Prestashop 1.6.1.24.

**Comment by musicpanda** [comment]
You could try my Prestools. It does some tests on the settings for the tree.

**Comment by Kos-tash** [comment, marked as answer]
While communicating with you, I decided to rewrite all the modules that were on the old version of the store and compare them with the new version. I received several modules that were only updated.
And I found the module through which all this happens. It can be Advanced Search 5 Pro.
It turns out that there is a setting called "Display products from subcategories" that causes these sorting changes. I turned it off and now everything works fine.

I decided to write it all down here because I've seen other users have the same problem. And usually the conversation ends without writing down what the problem was.

@musicpanda Thank you for helping me find the problem.
